crates/telio-proto/src/packet/pinger.rs

```diff
@@ -407,7 +416,10 @@ mod tests {
         assert_eq!(ping.encode().unwrap(), ping_bytes);
 
         let ping = PingerMsg::ping(WGPort(2), 3_u64, 10_u64);
-        let pong = ping.pong(WGPort(3), &"127.0.0.1".parse().unwrap()).unwrap();
+        let endpoint = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
+        let pong = ping
+            .pong(WGPort(3), &endpoint, EndpointProvider::Local)
+            .unwrap();
```

Comment:
Could you add unit tests for:
* Testing whether the EndpointProvider is correctly encoded as well as decoded. I would recommend hardcoding at least one message in byte array. And then asserting whether it encodes to that byte array, or decodes from it and fields are correct. This way we "backwards compatability" testing will be covered for later changes as well.
* Testing backwards compatability with old versions:
    * whether new versions can parse old byte arrays
    * whether old versions can parse new byte arrays